tests/src/cask_instance.rs

```diff
@@ -0,0 +1,329 @@
+#![allow(non_camel_case_types)]
+use std::collections::BTreeMap;
+
+use blake2::{
+    digest::{Update, VariableOutput},
+    VarBlake2b,
+};
+use casper_types::{bytesrepr::ToBytes, runtime_args, CLTyped, Key, RuntimeArgs, U256};
+use test_env::{Sender, TestContract, TestEnv};
+
+pub type TokenId = String;
+pub type Meta = BTreeMap<String, String>;
+pub type Gauge = BTreeMap<String, String>;
+pub type Warehouse = BTreeMap<String, String>;
+pub type Commission = BTreeMap<String, SubCommission>;
```

Comment:
We need to take a look if from the client we can actually push maps.
Also because as I recall we have plans to remove nested types from CLType (maps inside maps and such).